platform/ios/src/MGLMapView.mm

```diff
@@ -2388,7 +2392,8 @@ - (void)resetPosition
     auto camera = _mbglMap->getStyle().getDefaultCamera();
     CGFloat pitch = *camera.pitch;
     CLLocationDirection heading = mbgl::util::wrap(*camera.angle, 0., 360.);
-    CLLocationDistance distance = MGLAltitudeForZoomLevel(*camera.zoom, pitch, 0, self.frame.size);
+    CGFloat pitchForAltitude = self.isCameraAltitudeAffectedByPitch ? pitch : 0.0;
+    CLLocationDistance distance = MGLAltitudeForZoomLevel(*camera.zoom, pitchForAltitude, 0, self.frame.size);
     self.camera = [MGLMapCamera cameraLookingAtCenterCoordinate:MGLLocationCoordinate2DFromLatLng(*camera.center)
```

Comment:
Regardless of whether #12966 lands, we should probably change this to use 
`-[MGLMapCamera cameraLookingAtCenterCoordinate:altitude:pitch:heading:]` and rename the `distance` variable as `altitude`. Is my understanding correct?